File: Python/SpecHandphone.py
from Produk import Produk

class SpecHandphone(Produk):
    def __init__(self, id_produk="", nama="", harga=0,
                 processor="", ram=0, storage=0, battery=0):
        super().__init__(id_produk, nama, harga)
        self.__processor = processor
        self.__ram = ram
        self.__storage = storage
        self.__battery = battery

    def get_processor(self): return self.__processor
    def set_processor(self, processor): self.__processor = processor

    def get_ram(self): return self.__ram
    def set_ram(self, ram): self.__ram = ram

    def get_storage(self): return self.__storage
    def set_storage(self, storage): self.__storage = storage

    def get_battery(self): return self.__battery
    def set_battery(self, battery): self.__battery = battery

File: Java/SpecHandphone.java
public class SpecHandphone extends Produk {
    private String processor;
    private int ram;
    private int storage;
    private int battery;

    public SpecHandphone() {}

    public SpecHandphone(String idProduk, String nama, int harga,
                         String processor, int ram, int storage, int battery) {
        super(idProduk, nama, harga);
        this.processor = processor;
        this.ram = ram;
        this.storage = storage;
        this.battery = battery;
    }

    public String getProcessor() { return processor; }
    public void setProcessor(String processor) { this.processor = processor; }

    public int getRam() { return ram; }
    public void setRam(int ram) { this.ram = ram; }

    public int getStorage() { return storage; }
    public void setStorage(int storage) { this.storage = storage; }

    public int getBattery() { return battery; }
    public void setBattery(int battery) { this.battery = battery; }
}
